CMake/External_Darknet.cmake

```diff
@@ -0,0 +1,89 @@
+set(Darknet_DEPENDS)
+
+if(fletch_ENABLE_OpenCV)
```

Comment:
I don't know if there is a good reason not to..

Darknet seems kind of fragile and seems like a good idea to be able to access how it's built if somebody needs to so they can, for example, turn on CUDA and OpenCV off if for some reason the CUDA and OpenCV version don't get along or something to that effect

@mattdawkins could add 2 cents here

I am of the opinion to to expose all the flags available on these complicated codes, just in case....